intros/TypeScript for the Haskell or ML Programmer.md

```diff
@@ -0,0 +1,658 @@
+# TypeScript for Functional Programmers
+
+TypeScript began its life as an attempt to bring traditional object-oriented types
+to JavaScript so that the programmers at Microsoft could bring
+traditional object-oriented programs to the web. As it has developed, TypeScript's type
+system has evolved to model code written by native JavaScripters. The
+resulting system is powerful, interesting and messy.
+
+This introduction is designed for working Haskell or ML programmers
+who want to learn TypeScript. It describes how the type system of
+TypeScript differs from Haskell's type system. It also describes
+unique features of TypeScript's type system that arise from its
+modelling of JavaScript code.
+
+This introduction does not cover object-oriented programming. In
+practise, object-oriented programs in TypeScript are similar to those
+in other popular languages with OO features.
+
+# Prerequisites
+
+In this introduction, I assume you know the following:
+
+- How to program in JavaScript, the good parts.
+- Type syntax of a C-descended language.
+
+If you need to learn the good parts of JavaScript, read
+[JavaScript: The Good Parts](http://shop.oreilly.com/product/9780596517748.do).
+You may be able to skip the book if you know how to write programs in
+a call-by-value lexically scoped language with lots of mutability and
+not much else.
+[R<sup>4</sup>RS Scheme](https://people.csail.mit.edu/jaffer/r4rs.pdf) is a good example.
+
+[The C++ Programming Language](http://www.stroustrup.com/4th.html) is
+a good place to learn about C-style type syntax. Unlike C++, TypeScript uses postfix types, like so: `x: string` instead
+of `string x`.
+
+# Concepts not in Haskell
+
+## Built-in types
+
+JavaScript defines 7 built-in types:
+
+Type        | Explanation
+------------|-----------
+`Number`    | a double-precision IEEE 754 floating point.
+`String`    | an immutable 16-bit string.
```

Comment:
I think UTF-16 would be clearer than 16-bit, unless you intend some other meaning here

Reply:
Good point. Fixed.